### ISSUE TITLE
Move `loadImage()` into `Surface`, and AWT out of `PImage`

### ISSUE THREAD

**Issue by benfry** [benfry/processing4#56]
Several changes necessary to move image handling into `Surface` so that we're not invoking AWT. 

However, if that's done, we need to replace the image loading for JOGL and JavaFX as well, or have those subclass a LameKindaSurfaceAWT which has the AWT code. 

There's a constructor, resizer, and some IO options in `PImage` that all make use of AWT code. These will all be moved to `PImageJava2D` inside the `processing.awt` package, but will need replacements for JOGL and JavaFX.

**Comment by neilcsmith-net** [comment]
Are you trying to remove all AWT usage or just allow headless (for eg. LWJGL use)?  Because large portion of Processing AWT code, such as in PImage, PFont, etc. works fine headless.

**Comment by sampottinger** [comment]
Yeah I agree with @neilcsmith-net. There's some stuff like image saving where it's headless-OK but where AWT-dependent code has become standard. For example, it looks like even in [JavaFX](http://java-buddy.blogspot.com/2013/04/save-canvas-to-png-file.html) and [JOGL](https://stackoverflow.com/questions/22839899/how-to-save-am-image-of-a-screen-using-jogl), BufferedImage is still the canonical solution. There might not be a way to avoid using it.

**Comment by benfry** [comment]
More here: https://github.com/processing/processing4/wiki/Exorcising-AWT

**Comment by neilcsmith-net** [comment]
As per @sampottinger comment, LWJGL might be better placed in terms of library integration to allow a complete removal of AWT / java.desktop dependencies for a particular renderer.  @benfry interesting page, but surprised to see LWJGL still under discussion - thought that had been dropped?  My issue to discuss necessary changes to PApplet to support it were closed.
